### ISSUE TITLE
Current session unavailable - System.UnexpectedException

### ISSUE THREAD

**Issue by CloudJedi** [jongpie/NebulaLogger#97]
I'm trying to implement some logging in a scheduled job and got the following:

Any ideas what to check that could be causing this?

```
25915:20:00.0 (286295243)|METHOD_EXIT|[55]|01p3f000000TQ0F|Nebula.Logger.saveLog()
26015:20:00.0 (286460664)|FATAL_ERROR|System.UnexpectedException: Current session unavailable
261
262Class.Auth.SessionManagement.getCurrentSession: line 5, column 1
263Class.Nebula.LogEntryEventBuilder.getSessionMap: line 308, column 1
264Class.Nebula.LogEntryEventBuilder.setUserSessionDetails: line 258, column 1
265Class.Nebula.LogEntryEventBuilder.getLogEntryEvent: line 240, column 1
266Class.Nebula.Logger.saveLog: line 689, column 1
267Class.Nebula.Logger.saveLog: line 656, column 1
268Class.InstructorsEmailSender.execute: line 55, column 1
26915:20:00.0 (286694887)|CODE_UNIT_FINISHED| REMOVED
27015:20:00.0 (286723785)|EXECUTION_FINISHED
```

Sample:
```
global class InstructorsEmailSender implements Schedulable {
    global void execute(SchedulableContext SC) {
        Nebula.Logger.info('Running the InstructorsEmailSender Class');
        Nebula.Logger.saveLog();
    }
}

```

**Comment by CloudJedi** [comment]
I found the following code related to this:

```
    private Map<String, String> getSessionMap() {
        // Scheduled jobs, batch jobs, invocables, future methods and queueables are ALL
        // run by the Automated Process user - attempting to access Auth.SessionManagement
        // methods as that user throws an UNCATCHABLE error. Let's sidestep that.
        String potentialSession = UserInfo.getSessionId();
        if (String.isBlank(potentialSession) || Test.isRunningTest()) {
            return new Map<String, String>{ 'SessionId' => potentialSession };
        }

        return Auth.SessionManagement.getCurrentSession();
    }
```

**Comment by jongpie** [comment]
@CloudJedi thanks for reporting this! We made a few changes for this a couple of months ago to try to solve these types of issues (PR #66), but it looks like it's not fully resolved for scheduled jobs. 

I think I can change the code to query the [AuthSession object](https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_authsession.htm) instead of using `Auth.SessionManagement.getCurrentSession();` I'll work on a fix for this over the weekend - if everything goes well, I'll merge the change to `master` and will include the fix in the next managed package release, [v4.1.0](https://github.com/jongpie/NebulaLogger/milestone/2).

Let me know if you run into any other issues!

**Comment by jamessimone** [comment]
@CloudJedi sorry about that. I tested the fix for #66 out with batch jobs and queueables, but not with a scheduable!

**Comment by jongpie** [comment]
@CloudJedi I was able to wrap up a fix for this as a part of my pull request #96. I'm planning to release the next managed package version hopefully this weekend, which will include this bugfix.

**Comment by CloudJedi** [comment]
Awesome looking forward to trying out the new version!
